include/libdnf/rpm/sack.hpp

```diff
@@ -61,11 +68,53 @@ inline ReldepId::ReldepId(int id) : id(id) {}
 class Package;
 class Reldep;
 class ReldepList;
+class Repo;
 
 class Sack {
 public:
-    Sack();
+    class Exception : public RuntimeError {
+    public:
+        using RuntimeError::RuntimeError;
+        const char * get_domain_name() const noexcept override { return "libdnf::rpm::Sack"; }
+        const char * get_name() const noexcept override { return "Exception"; }
+        const char * get_description() const noexcept override { return "rpm::Sack exception"; }
+    };
+
+    class NoCapability : public Exception {
+    public:
+        using Exception::Exception;
+        const char * get_domain_name() const noexcept override { return "libdnf::rpm::Sack"; }
+        const char * get_name() const noexcept override { return "NoCapability"; }
+        const char * get_description() const noexcept override {
+            return "repository does not provide required metadata type";
+        }
+    };
+
+    // DnfSackAddFlags:
+    // @DNF_SACK_LOAD_FLAG_NONE:                    No flags set
+    // @DNF_SACK_LOAD_FLAG_BUILD_CACHE:             Build the solv cache if required
+    // @DNF_SACK_LOAD_FLAG_USE_FILELISTS:           Use the filelists metadata
+    // @DNF_SACK_LOAD_FLAG_USE_PRESTO:              Use presto deltas metadata
+    // @DNF_SACK_LOAD_FLAG_USE_UPDATEINFO:          Use updateinfo metadata
+    // @DNF_SACK_LOAD_FLAG_USE_OTHER:               Use other metadata
+    //
+    // Flags to use when loading from the sack.
+    typedef enum {
```

Comment:
I believe that it could be replaced by enum class. So far it is used by repo_impl->libsolv_repo_ext.load_flags. The question is how data are passed to librepo or to libsolv?

Reply:
Please what relation between DataType and DnfSackLoadFlags?

Reply:
An enum class will be better for clean C++ interface.  But it is public API. How we export it to plain C?
Anyway. The presented solution is also problematic to export. I've ideas. We must discuss it.

Reply:
@j-mracek I created enum class. And done some renaming. I hope it is more clear now.